bot/utils/dma_api.py

```diff
@@ -47,31 +47,54 @@ async def wrapped(search):
     return wrapper
 
 
-@get_api()
-async def get_article(number, articles):
-    return [
-        item
-        for item in articles
-        if item["series"] == str(number)[0]
-        and item["category"] == str(number)[1]
-        and item["digit"] == str(number)[2]
-    ]
-
-
-@get_api()
-async def get_series(series, articles):
-    return [item for item in articles if item["series"] == str(series)]
+async def get_search_api(query_string: str = None, series_number: int = None):
+    query = "?query={" + re.sub(DMA_QUERY_REGEX, "", query_string) + "}" if query_string is not None else ""
+    series = "?section=" + DMA_SERIES_IDS[series_number] if series_number is not None else ""
+    request_url = f'{DMA_API_SEARCH_URL}{query}{series}'
```

Comment:
A refactor of this could use urllib to build these request strings to avoid the concatenation and long lines:
```py
In [1]: from urllib import parse

In [2]: url_params = {
   ...:   "section": 4
   ...: }

In [3]: parse.urlencode(url_params)
Out[3]: 'section=4'
```

Additionally, `is not None` is redundant here which can reduce the line length.

```py
In [4]: series_number = None

In [5]: "A" if series_number else "B"
Out[5]: 'B'

In [6]: series_number = 4

In [7]: "A" if series_number else "B"
Out[7]: 'A'
```